src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/EndpointContracts/DocumentHighlightParamsBridge.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using MediatR;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+
+/// <summary>
+/// This class is used as a "bridge" between the O# and VS worlds. Ultimately it only exists because the base <see cref="RenameParams"/>
+/// type does not implement <see cref="IRequest"/>.
+/// </summary>
+/// <remarks>
+/// Normally we would inherit this from DocumentHighlightParams, which in turn inherits from TextDocumentPositionParams
+/// but in this case, we can't do that, we have to go direct.
+/// The reason we can't inherit from DocumentHighlightParams is because it uses IProgress, and IProgress is not
+/// supported in the middle layer. No, I don't know what "the middle layer" is either. I guess we're in
+/// the middle layer right now? No idea. Taylor probably knows, I'm just guessing based on this comment:
+/// <![CDATA[
+/// https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=/src/product/RemoteLanguage/Impl/JsonRpcExtensionMethods.cs&version=GBdevelop&line=142&lineEnd=143&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
```

Comment:
- To avoid future broken links, should we link to a commit instead? e.g. https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?version=GC2d1fe0c5ab668d49b0404bed9cd658a78f696165&path=/src/product/RemoteLanguage/Impl/JsonRpcExtensionMethods.cs
- Should we file a work item on the client side to track support for IProgress serialization? (Although I'm not sure how big of a task this is)

Reply:
> Should we file a work item on the client side to track support for IProgress serialization?

This needs more investigation as to who needs to change what, and how. Our "middle layer" doesn't do anything, but seems like if I remove it entirely, this still doesn't work, so there could also be something else going on.

This will have to be addressed when we convert an endpoint that actually uses IProgress (FAR for example) so figured we'd leave it til then. Roslyn doesn't stream results for highlights, so it's safe to ignore here

Reply:
> To avoid future broken links, should we link to a commit instead?

Fixed